### ISSUE TITLE
User Groups for Permissions and Exploration of Notes

### ISSUE THREAD

**Issue by InnayTool** [hedgedoc/hedgedoc#999]
**Idea:**
We introduce the concept of groups in the development of hedgedoc 2.0 already in the backend.
Groups are only capable of intern special groups by now `loggedIn` and `everyone` (Issue #908).
This can be extended to groups where a set of users can be added.

**Use-cases:**
- Set specific permissions for users on a note
- Groups can own a note (Issue #55 -> Issue #998)
- List all notes a group has access if user is in this group (Issue #54)
	- This is related to Explore Page #994 

**Details already discussed:**
The 2.0 backend is capable of groups with a set of users by now.
There is also a quite complex structure of options for anonymous edits (Issue #856)

**Comment by funkyfuture** [comment]
yes. this is one point why we didn't adopt (then) CodiMD at our organization.

since pads are often used to work ad-hoc on contents, it would be neat to allow all users to also create and edit groups by themselves. sure, the availability of this feature might be configurable at the instance level.

shouldn't the first bullet point read: "Set specific permissions for a user group on a note" ?

**Comment by InnayTool** [comment]
By now it seems to be a wise decision to drop this feature for 2.0 and move it to a later version to get 2.0 in a usable state.
